CodeParser/CodeParser.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <fstream>
+#include <filesystem>
+#include <regex>
+#include <string>
+
+using namespace std;
+
+class CodeParser
+{
+public:
+	CodeParser() : m_blank_lines_count(0),
+		m_comment_lines_count(0), m_code_lines_count(0) {};
+	CodeParser(string file_path) : m_file_path(file_path), m_blank_lines_count(0),
+	m_comment_lines_count(0), m_code_lines_count(0) {};
+	~CodeParser() {};
+
+	void Parse(string file_path);
```

Comment:
use reference here for a string

Reply:
> use reference here for a string

In this case i pass string as a value, because function 'Parse()' is passed to thread.
When string is pass as a reference i need to write std::ref(path) to pass it as a thread parametr, but 'path' is the returned value from the other function.
Compiler does not allow writing 
thread(&CodeParser::Parse, m_parsers[i], std::ref(reader.GetPath(i)));
Cause it i must to write
string path = reader.GetPath(i);
thread(&CodeParser::Parse, m_parsers[i], std::ref(path));

It is more then 1 operation and in this case is needed to use some mutex for synhronization.
When synhronization is not used will be incorect result. 
In this case, it is more profitable to pass string as the value

Reply:
I tryed to pass string as a referens, but got incorrect results. Cause it pas string as a value